### PR TITLE
Simplifying paths normalization and resolving

### LINKED ISSUE #16: Allow colon in filenames
The colon symbol is currently in the set of illegal characters for file names:

https://github.com/in-toto/in-toto-rs/blob/2b9ba422fe079395cb01b8a18d5da5df0b980a2d/src/models/helpers.rs#L54-L55

This prevents in-toto from generating signatures for Arch Linux packages that have an epoch, eg `rust-1:1.56.1-3-x86_64.pkg.tar.zst`.

### PR DESCRIPTION
**Pull request by Mo-Fatah**
This PR fixes #16  with addressing some linting warning form Clippy.

### CODE REVIEW

**Comment by Mo-Fatah** [comment]
Should I disable Clippy for the warned lines or just edit them to make Clippy happy ?

**Review by SantiagoTorres — COMMENTED** [review]
I like this!

I think there is a minor nit, but I'm happy with how the code became way simpler :)

File: src/models/helpers.rs [hunk 1]
```diff
@@ -143,87 +21,18 @@ pub struct VirtualTargetPath(String);
...
-                    .collect::<HashSet<_>>()
-            })
-            .collect::<Vec<_>>();
-        self.matches_chain(&*new)
+        //safe_path(&path)?;
+        let cleaned_path = match canonicalize_path(Path::new(path.as_str())) {
+            Ok(path_buf) => {
+                match path_buf.to_str() {
+                   Some(x) => x.to_string(),
+                   None => " ".to_string() // TODO
```
[the remark this answers is not in the document]
Comment (Mo-Fatah) [reply src/models/helpers.rs:30]: Yeah. I wasn't sure what error exactly to raise. I think [VerificationFailure](https://github.com/in-toto/in-toto-rs/blob/11c7a419dcfece2f3e584a08f93ecbb1d74d69e8/src/error.rs#L55) would be suitable ?

[the remark this answers is not in the document]
Comment (SantiagoTorres) [reply src/models/helpers.rs]: Let's leave it for now. VerificationFailure should be done for verifylib bits, so I'm not sure

**Comment by SantiagoTorres** [comment]
Thank you @Mo-Fatah !  I"ll merge now, and we can hold off for a couple of more features before cutting a release.

**Pull request merged**